ui/app/components/dashboard/quick-actions-card.js

```diff
@@ -0,0 +1,182 @@
+/**
+ * Copyright (c) HashiCorp, Inc.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+
+import Component from '@glimmer/component';
+import { action } from '@ember/object';
+import { tracked } from '@glimmer/tracking';
+import { inject as service } from '@ember/service';
+
+/**
+ * @module DashboardQuickActionsCard
+ * DashboardQuickActionsCard component allows users to see a list of secrets engines filtered by
+ * kv, pki and database and perform certain actions based on the type of secret engine selected
+ *
+ * @example
+ * ```js
+ * <DashboardQuickActionsCard />
+ * ```
+ * @param {array}
+ */
+
+const getActionsByEngineType = (type) => {
+  switch (type) {
+    case 'kv':
+      return [
+        {
+          actionTitle: 'Find KV secrets',
+          actionType: 'find-kv',
+        },
+      ];
+    case 'database':
+      return [
+        {
+          actionTitle: 'Generate credentials for database',
+          actionType: 'generate-credentials-db',
+        },
+      ];
+    case 'pki':
+      return [
+        {
+          actionTitle: 'Issue certificate',
+          actionType: 'issue-certificate-pki',
+        },
+        {
+          actionTitle: 'View certificate',
+          actionType: 'view-certificate-pki',
+        },
+        {
+          actionTitle: 'View issuer',
+          actionType: 'view-issuer-pki',
+        },
+      ];
+  }
+};
+
+const getActionParamByAction = (type) => {
+  switch (type) {
+    case 'find-kv':
+      return {
+        title: 'Secret Path',
```

Comment:
I noticed that on initial load it says "Secret Path" but changes when I click database which then stays on the next title after I delete the selected mount. Looks like we need to reset the action selection when the mount is removed. This kind of goes with my comment above about "Find KV secrets". We should default back to some default or empty action, which shows on load and whenever a mount is removed.

Reply:
I can't reproduce that issue 🤔 I might have reproduced it wrong - please see the video below :)

https://github.com/hashicorp/vault/assets/30884335/ad7bf5af-f470-4030-b7e9-1598fb657184